File: backend/instancia-3/src/main/java/br/imd/ufrn/atendimento/service/AnalystService.java
package br.imd.ufrn.atendimento.service;

import br.imd.ufrn.atendimento.dto.AnalystRequest;
import br.imd.ufrn.atendimento.dto.AnalystResponse;
import java.util.List;

public interface AnalystService {

    AnalystResponse create(AnalystRequest request);

    AnalystResponse findById(Long id);

    List<AnalystResponse> findAll();

    AnalystResponse update(Long id, AnalystRequest request);

    void delete(Long id);
}

File: frontend/instancia-3/src/app/services/analyst.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { AnalystRequest, AnalystResponse } from '../models/analyst.model';

@Injectable({ providedIn: 'root' })
export class AnalystService {
  private readonly http = inject(HttpClient);
  private readonly base = '/api/v1/analysts';

  list(): Observable<AnalystResponse[]> {
    return this.http.get<AnalystResponse[]>(this.base);
  }

  findById(id: number): Observable<AnalystResponse> {
    return this.http.get<AnalystResponse>(`${this.base}/${id}`);
  }

  create(request: AnalystRequest): Observable<AnalystResponse> {
    return this.http.post<AnalystResponse>(this.base, request);
  }

  update(id: number, request: AnalystRequest): Observable<AnalystResponse> {
    return this.http.put<AnalystResponse>(`${this.base}/${id}`, request);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.base}/${id}`);
  }
}
